_pytest/cacheprovider.py

```diff
@@ -161,8 +156,9 @@ def pytest_sessionfinish(self, session):
         config = self.config
         if config.getvalue("cacheshow") or hasattr(config, "slaveinput"):
             return
-        prev_failed = config.cache.get("cache/lastfailed", None) is not None
-        if (session.testscollected and prev_failed) or self.lastfailed:
+
+        saved_lastfailed = config.cache.get("cache/lastfailed", {})
```

Comment:
Not sure what you mean, could you clarify? Also not sure if it was just a general comment or a request for change.